File: sport-team-manager-frontend/src/app/models/match.ts
import { Player } from "./player";

export interface Match {
  id: string;
  date: string;
  opponent: string;
  location: string;
  matchType: string;
  teamScore?: number;
  opponentScore?: number;
  result: string;
  notes?: string;
  teamId: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateMatch {
  date: string;
  opponent: string;
  location: string;
  matchType: string;
  teamScore?: number;
  opponentScore?: number;
  notes?: string;
  teamId: string;
}

export interface UpdateMatch {
  date: string;
  opponent: string;
  location: string;
  matchType: string;
  teamScore?: number;
  opponentScore?: number;
  result: string;
  notes?: string;
}

// ✅ CORRIGIDO: MatchWithPlayers deve extender Match e adicionar matchPlayers
export interface MatchWithPlayers extends Match {
  matchPlayers: MatchPlayer[];
}

export interface MatchPlayer {
  id: string;
  matchId: string;
  playerId: string;
  isStarter: boolean;
  position?: string;
  minutesPlayed?: number;
  goals?: number;
  assists?: number;
  yellowCards?: number;
  redCards?: number;
  notes?: string;
  player?: Player;
}

File: backend/SportTeamManager.Domain/Entities/Match.cs
namespace SportTeamManager.Domain.Entities;

public class Match
{
    public Guid Id { get; set; }
    public DateTime Date { get; set; }
    public string Opponent { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public string MatchType { get; set; } = "Friendly"; // Friendly, League, Cup, etc.
    public int? TeamScore { get; set; }
    public int? OpponentScore { get; set; }
    public string Result { get; set; } = "Scheduled"; // Win, Loss, Draw, Scheduled
    public string? Notes { get; set; }
    public Guid TeamId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public virtual Team Team { get; set; } = null!;
    public virtual ICollection<MatchPlayer> MatchPlayers { get; set; } = new List<MatchPlayer>();
}